assembly/std/address.ts

```diff
@@ -29,81 +29,55 @@ export class Address implements Valider {
   }
 
   /**
-   * Returns the offset of the next element after having parsed
-   * an address from a string segment.
+   * Serialize the address
    *
-   * The string segment can contains more thant on serialized element.
+   * @remarks
+   * Addresses are not fixed-size so the first bytes are the size as a i32, then the address string is encoded.
    *
-   * @param bs -
-   * @param begin -
+   * @returns the bytes
    */
-  fromStringSegment(bs: string, begin: i32 = 0): i32 {
-    const length = u8(bs.codePointAt(begin));
-    // return length;
-    this._value = Address.fromByteString(
-      bs.slice(begin + 1, begin + length + 1),
-    ).toByteString();
-    return begin + length + 1;
+  serialize(): StaticArray<u8> {
+    return new Args().add(this._value).serialize();
   }
 
   /**
-   * Returns a string segment.
+   * Deserialize the address
    *
-   * The string segment can be concatenated with others
-   * to serialize multiple elements.
+   * @param data - bytes
+   * @param offset - `Args` instance current offset
+   * @returns the new offset wrapped in a `Result`
    */
-  toStringSegment(): string {
-    return String.fromCharCode(u8(this._value.length)).concat(
-      this.toByteString(),
-    );
-  }
-
-  /**
-   * Returns an Address from a byte string.
-   *
-   * @param bs - Byte string
-   */
-  static fromByteString(bs: string): Address {
-    return new Address(bs);
+  deserialize(data: StaticArray<u8>, offset: i32 = 0): Result<i32> {
+    const args = new Args(data, offset);
+    const result = args.nextString();
+    if (result.isErr()) {
+      return new Result(0, "Can't deserialize address.");
+    }
+    this._value = result.unwrap();
+    return new Result(args.offset);
   }
 
   /**
    * Serialize to byte string.
    */
-  toByteString(): string {
+  toString(): string {
```

Comment:
is it a big breaking change ?

Reply:
no because only the name of the function changes, not the logic